snuba/datasets/factory.py

```diff
@@ -42,3 +44,9 @@ def get_dataset(name):
 
 def get_enabled_dataset_names():
     return [name for name in DATASET_NAMES if name not in settings.DISABLED_DATASETS]
+
+
+def enforce_table_writer(dataset: Dataset) -> TableWriter:
```

Comment:
This is just a utility method to ensure the developers who try to shoot themselves in the feet by assuming all datasets have a TableWriter go through a method that claims that's a wrong assumption. So that we do not have to copy paste the assertion everywhere